### DISCUSSION TITLE
If I only use a package inside of `getStaticProps` will it be included in the browser's bundle?

### DISCUSSION THREAD

**Discussion by ricokahler** [vercel/next.js#14568] [Help] [answered]
If I only use a package inside of `getStaticProps`, will it be included in the browser's bundle?

e.g. 

```js
import someLargePackage from 'some-large-package';

export async function getStaticProps(context) {
  const something = await someLargePackage(context.params?.slug);

  return {
    props: { something },
  }
}

// ...
```

This some-large-package is really only needed during build time. I'm curious if next knows how to remove it from browser bundle.

Thanks!

**Comment by Timer** [comment, marked as answer]
This will not be in the browser bundle:
https://nextjs.org/docs/basic-features/data-fetching#write-server-side-code-directly

You can put your code in this to verify:
https://next-code-elimination.now.sh/

**Reply by ricokahler** [reply]
I tried using [`sharp`](https://github.com/lovell/sharp) to convert images within `getStaticProps` and it gave me an error saying something about how sharp uses `child_process`. Am I able to use sharp in `getStaticProps`?

**Reply by Timer** [reply]
Yes. Make sure the code is eliminated in that tool above!
